os/pm/pm_addhistory.c

```diff
@@ -0,0 +1,95 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdlib.h>
+#include <string.h>
+#include <queue.h>
+#include <debug.h>
+#include <time.h>
+#include <assert.h>
+#include <tinyara/pm/pm.h>
+#include "pm.h"
+#include "pm_metrics.h"
+
+#ifdef CONFIG_PM
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pm_addhistory
+ *
+ * Description:
+ *   This function add history into pm structure.
+ *
+ * Input parameters:
+ *   domain - The PM domain to check
+ *   state - Current PM state
+ *
+ * Returned value:
+ *    None.
+ *
+ ****************************************************************************/
+#ifdef CONFIG_PM_METRICS
+void pm_addhistory(int domain, int state)
```

Comment:
This is one of kernel feature, you had better modify type to int, instead of void.
Then you can return error values which is defined in errno.h